core-tests/shared/src/test/scala/zio/ZIOSpec.scala

```diff
@@ -2115,6 +2149,40 @@ object ZIOSpec extends ZIOBaseSpec {
         }
       }
     ),
+    suite("validateM")(
+      testM("returns all errors if never valid") {
+        val in  = List.fill(10)(0)
+        val res = ZIO.validateM[Any, Int, Int, Int](in)(a => ZIO.fail(a)).flip
```

Comment:
Don't think we should need the type annotations here. If you get a warning for a type being inferred as `Any` it is just a spurious warning for the environment type that you can silence by using `IO.validateM`.

Reply:
Still need to do this.